### PR TITLE
Add a way to specify emulator host/port for Auth

### PR DESCRIPTION
**Pull request by avolkovi**

### CODE REVIEW

**Review by MeghaB — APPROVED** [review]
lgtm - thanks for adding the console warning!

File: packages/auth/src/auth.js [hunk 1]
```diff
@@ -272,6 +292,63 @@ fireauth.Auth.prototype.useDeviceLanguage = function() {
...
+
+/**
+ * Emits a console warning and a visual banner if emulator integration is
+ * enabled.
+ */
+fireauth.Auth.prototype.emitEmulatorWarning_ = function() {
+  fireauth.util.consoleWarn('WARNING: You are using the Auth Emulator,' +
+    ' which is intended for local testing only.  Do not use with' +
+    ' production credentials.');
+  if (goog.global.document) {
```
Comment (yuchenshi) [inline packages/auth/src/auth.js:324]:
```suggestion
  if (goog.global.document && goog.global.document.body) {
```

File: packages/auth/src/auth.js [hunk 2]
```diff
@@ -272,6 +292,63 @@ fireauth.Auth.prototype.useDeviceLanguage = function() {
...
+/**
+ * Emits a console warning and a visual banner if emulator integration is
+ * enabled.
+ */
+fireauth.Auth.prototype.emitEmulatorWarning_ = function() {
+  fireauth.util.consoleWarn('WARNING: You are using the Auth Emulator,' +
+    ' which is intended for local testing only.  Do not use with' +
+    ' production credentials.');
+  if (goog.global.document && goog.global.document.body) {
+    const ele = goog.global.document.createElement('p');
```
Comment (samhorlbeck) [inline packages/auth/src/auth.js:325]: This should probably be a `<div>`instead but no need to change it now

**Review by samhorlbeck — APPROVED** [review]

**Review by Feiyang1 — CHANGES_REQUESTED** [review]
Can you please add the corresponding typings to `packages/firebase/index.d.ts`?

**Comment by Feiyang1** [comment]
Actually are you planning to make this API public now, or later?

File: packages/firebase/index.d.ts [hunk 3]
```diff
@@ -3119,6 +3119,14 @@ declare namespace firebase.auth {
      * Sets the current language to the default device/browser preference.
      */
     useDeviceLanguage(): void;
+    /**
+     * Modify this Auth instance to communicate with the Firebase Auth emulator.  This must be 
+     * called synchronously immediately following the first call to `firebase.auth()`.  Do not use
+     * with production credentials as emulator traffic is not encrypted.
+     * 
+     * @param url - The URL at which the emulator is running (eg, 'http://localhost:9099')
```
Comment (Feiyang1) [inline packages/firebase/index.d.ts:3127]:
```suggestion
     * @param url The URL at which the emulator is running (eg, 'http://localhost:9099')
```

We are using JSDoc currently, which doesn't requires the `-` after the  parameter name

**Review by Feiyang1 — APPROVED** [review]

**Pull request merged**